spec/solver_spec.rb

```diff
@@ -0,0 +1,20 @@
+require './solver'
+
+describe 'Solver' do
+  context 'Testing class Solver' do
+    it 'Should return factorial of a number, reverse of a string and  print fizz, buzz and fizzbuzz' do
+      solver_mock = double('solver')
+      solver_class = Solver.new
+      allow(solver_mock).to receive(:factorial).with(3).and_return(6)
+      allow(solver_mock).to receive(:reverse).with('hello').and_return('olleh')
+      allow(solver_mock).to receive(:fizzbuzz).with(3).and_return('fizz')
+      allow(solver_mock).to receive(:fizzbuzz).with(5).and_return('buzz')
+      allow(solver_mock).to receive(:fizzbuzz).with(15).and_return('fizzbuzz')
+      expect(solver_class.factorial(3)).to eq(6)
+      expect(solver_class.reverse('hello')).to eq('olleh')
+      expect(solver_class.fizzbuzz(3)).to eq('fizz')
+      expect(solver_class.fizzbuzz(5)).to eq('buzz')
+      expect(solver_class.fizzbuzz(15)).to eq('fizzbuzz')
```

Comment:
- [ ] For `fizzbuzz` method, Kindly add an edge case when the number is not divisible by either 3 or 5, To cover all test cases.

Reply:
Hello @Omar-Muhamad thank you for the remarks, we will correct and resubmit